packages/amplify-codegen/src/utils/getOutputFileName.js

```diff
@@ -1,18 +1,27 @@
 const path = require('path');
 
-function getOutputFileName(inputFileName = 'API', target) {
+function getOutputFileName(inputFileName, target) {
   const extensionMap = {
     swift: 'swift',
     typescript: 'ts',
     flow: 'js',
+    angular: 'service.ts',
   };
+  const folderMap = {
+    javascript: 'src',
+    typescript: 'src',
+    flow: 'src',
+    angular: 'src/app',
+  };
+
+  inputFileName = inputFileName || target === 'angular' ? 'api' : 'API';
```

Comment:
Can you explain what this is doing? and in what cases target = api vs API? This check seems a bit fragile.

Reply:
For every other language other then angular we are using `API` as the base file name. For angular its lowercase. Also the first check is if the user has already specified the `inputFileName` we provide no default file name